pages/form/index.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { FormEvent, useState } from "react";
+import axios from "axios";
+import InputField from "../../components/registration-form/InputField";
+
+/**
+ * Types
+ */
+
+export type FieldState = {
+  value: string;
+  error: string | undefined;
+};
+
+export type FormState = {
+  [fieldName: string]: FieldState;
+};
+
+type FieldDescription = {
+  type: "text" | "password";
+  label: string;
+  placeholder?: string;
+};
+
+/**
+ * Constants
+ */
+
+const fields: { [key: string]: FieldDescription } = {
```

Comment:
can you add field-level validations here?

then group all these into a `useForm` React hook

A hook in this case is really just a function that groups similar functionalities together.

```
function useForm() {
  const [formState, setFormState] =  useState(...);
  const updateFieldValue = () => {...};
  return {
    formState,
    updateFieldValue,
    ... // and more
}
```